### PR TITLE
Try fix log reconfigure error

### PR DESCRIPTION
**Pull request by jilen**
# Pull Request Checklist

## Fixes

Try fix ci builds

## Note
The `SLF4JBrigerHandler` will modify the `LoggerContext` concurrently in `shutdown` and `configure` method.

### CODE REVIEW

**Comment by jilen** [comment]
Build passed successfully... But not sure it is fixed by this pr or just success randomly.

**Comment by mkurz** [comment]
Thanks! At first glance this PR makes total sense. Will take a closer look asap.

> Build passed successfully... But not sure it is fixed by this pr or randomly.

We will see in upcoming PRs :wink:

**Comment by mkurz** [comment]
@jilen This change does _not_ fix the problem. I restarted the tests to see if the exception occurs again and yes it does...

**Comment by jilen** [comment]
> @jilen This change does _not_ fix the problem. I restarted the tests to see if the exception occurs again and yes it does...
Hm, I'll try to identify the root cause.

File: build.sbt [hunk 1]
```diff
@@ -339,6 +339,7 @@ lazy val PlayIntegrationTestProject = PlayCrossBuiltProject("Play-Integration-Te
     (Test / parallelExecution)   := false,
     mimaPreviousArtifacts        := Set.empty,
     (Test / fork)                := true,
+    (Test / testOptions) += Tests.Argument("sequential"),
```
Comment (mkurz) [inline build.sbt:342]: Actually this is just a workaround but does not fix the original problem.
IMHO the high level problem is that the logging initialization and "closing" of the logger is using static  global method/vars. Now when a logger gets created another thread can just afterwards close it so the code that wants to access the logger will run into a null pointer exception. 
This is just my idea by scrolling through the code.
I think the real solution is to get rid of the static, global code.

Comment (jilen) [reply build.sbt:342]: Very likely related to the `slf4j` / `logback` static initialization process.

**Comment by jilen** [comment]
@mkurz Seems fixed.

**Comment by mkurz** [comment]
I restart the tests a couple of times to check if we run into it again. On first sight your changes makes sense.

(Side note) However, not sure if in the end, in a different PR, more works needs to be done to correctly run tests in parallel with "correct" logging set up... Right now still statics are used and therefore parellel tests can cause `configure` to run multiple times, overriding global vars, same with the `shutdown` method. However didn't look into that yet in detail.

**Review by mkurz — APPROVED** [review]
Your changes definitely make sense. Even if not fixing the logger related errors, synchronizing this code should not hurt.

:crossed_fingers: Hopefully this fixes the random CI errors related to logging.

**Comment by mkurz** [comment]
@Mergifyio backport 3.0.x

**Comment by mkurz** [comment]
@Mergifyio backport 2.9.x

**Comment by mkurz** [comment]
Looks like this really fixed the random CI errors that occured quite often (sometimes on each run). Since merging this PR many other PR's were opened / merged and the CI error related to logging did not happen again.
Thanks for taking time to look into this @jilen!

**Pull request merged**